src/actinia_core/core/common/app.py

```diff
@@ -136,10 +140,15 @@
     consumes=["application/gml+xml", "application/json"],
 )
 
-# Set the security definition in an unconventional way
-flask_api._swagger_object["securityDefinitions"] = {
-    "basicAuth": {"type": "basic"}
-}
-flask_api._swagger_object["security"] = [{"basicAuth": []}]
-
-auth = HTTPBasicAuth()
+# authentication method
+global_config.read(DEFAULT_CONFIG_PATH)
+if global_config.KEYCLOAK_CONFIG_PATH:
+    auth = HTTPTokenAuth(scheme="Bearer")
```

Comment:
Maybe it is possible to add flask description for bearer auth as well? (https://swagger.io/specification/v2/#securityDefinitionsObject)